### DISCUSSION TITLE
Encoder missing

### DISCUSSION THREAD

**Discussion by ermoschink** [donmccurdy/glTF-Transform#1409] [Q&A] [answered]
I'm trying to optimize the meshes.

```
                          await document.transform(
                              // Losslessly resample animation frames.
                             /* resample(),
                              // Remove unused nodes, textures, or other data.
                              prune(),
                              // Remove duplicate vertex or texture data, if any.
                              dedup(),
                              // Compress mesh geometry with Draco.
                              draco(),*/
                              meshopt({encoder: MeshoptEncoder, level: 'medium'}),
                              // Convert textures to WebP (Requires glTF Transform v3 and Node.js).

                              // Custom transform.
                          );
```
And I get a crash on the lines **encoder.encodeGltfBuffer(.....)** so the encoder is undefined.
```
      const srcArray = BufferUtils.concat(bufferViewData);
      const dstArray = encoder.encodeGltfBuffer(srcArray, count, byteStride, mode);
```

Please tell me what I missed?

**Comment by donmccurdy** [comment, marked as answer]
See https://gltf-transform.dev/modules/extensions/classes/EXTMeshoptCompression. I would check that:

1. `MeshoptEncoder` is imported as shown
2. `MeshoptEncoder.ready` is awaited
3. `MeshoptEncoder` is passed both to the meshopt() function, and to the I/O class dependencies

Aside – there is little point in applying both Draco and Meshopt, both cannot be applied to the same mesh. You probably will want to choose one of the two.

**Reply by ermoschink** [reply]
Thank you!
I don't intend to use both)
